src/app/zap-templates/call-command-handler-src.zapt

```diff
@@ -100,15 +100,64 @@ EmberAfStatus emberAf{{asCamelCased name false}}Cluster{{asCamelCased side false
         {{#if (isCommandAvailable parent.side incoming outgoing commandSource name)}}
         case ZCL_{{asDelimitedMacro name}}_COMMAND_ID: {
             {{#if (zcl_command_arguments_count this.id)}}
-            uint32_t argOffset = 0;
+            uint32_t payloadOffset = cmd->payloadStartIndex;
             {{#zcl_command_arguments}}
-            {{asUnderlyingType type}} * {{asSymbol label}} = ({{asUnderlyingType type}} *)(cmd->buffer + argOffset);
+            {{#if (isNullable type isArray)}}
+            uint8_t * {{asSymbol label}};
+            {{else if (isStrEqual label "groupId")}}
+            GroupId {{asSymbol label}};
+            {{else}}
+            {{asUnderlyingType type}} {{asSymbol label}};
+            {{/if}}
+            {{/zcl_command_arguments}}
+
+            {{#zcl_command_arguments}}
+            {{#if presentIf}}
+            if ({{presentIf}})
+            {
+            {{/if}}
+            {{#if isArray}}
+            {{asSymbol label}} = (uint8_t *)(cmd->buffer + payloadOffset);
+            {{else if (isString type)}}
+            {{asSymbol label}} = emberAfGetString(cmd->buffer, payloadOffset, cmd->bufLen);
+            {{else if (isLongString type)}}
+            {{asSymbol label}} = emberAfGetLongString(cmd->buffer, payloadOffset, cmd->bufLen);
+            {{else if (isStrEqual label "groupId")}}
+            {{asSymbol label}} = (*(GroupId *)(cmd->buffer + payloadOffset));
+            {{else}}
+            {{asSymbol label}} = (*({{asUnderlyingType type}} *)(cmd->buffer + payloadOffset));
```

Comment:
This is all sorts of broken, including missing length checks and endianness fail, but I guess once we move to `DataModelReader` this will be moot....

Maybe add a comment that includes the string "emberAfGetInt" in it so when I search for that I find this spot?

Reply:
Yeah...

The tricky thing is that `{{asUnderlyingType}}` is an `async` method and I have not found the right incantation that would have allowed me to pass it to a `helper` function.

Otherwise I could easily have written an `{{#if}}` case for all types, like:
```
{{#if (isStrEqual {{asunderlyingType type}} "uint8_t")}}
{{asSymbol label}} = emberAfGetInt8u(cmd->buffer, payloadOffset, cmd->bufLen);
payloadOffset += 1u;
{{else if (isStrEqual {{asunderlyingType type}} "uint16_t")}}
{{asSymbol label}} = emberAfGetInt16u(cmd->buffer, payloadOffset, cmd->bufLen);
payloadOffset += 2u;
{{ else if ...}}
{{/if}}
if (cmd->Buffer + payloadOffset >= cmd->bufLen)
{
    return EMBER_ZCL_STATUS_MALFORMED_COMMAND;
}
```

And maybe I could have used an `accumulator` (https://github.com/project-chip/zap/blob/72c4b75847d92d077e0be02cb8fba14a4900a830/test/gen-template/zigbee/accumulator.zapt#L8) at the beginning of the function to

But yes. In general this code is clearly unsafe.

Reply:
@vivien-apple I'm concern about the stack usage of your new code. The goal of using pointer to the data was to prevent unnecessary stack usage since all of those variables will be pass directly to the callback handler. Seems to me like this use more stack and is less scalable than the previous implementation.

Reply:
I'm not sure about the less scalable part but for sure it will use more stack. The current plan for this code is to be rewritten using `DataModelReader` (https://github.com/project-chip/connectedhomeip/blob/84847dd65c9ba9b0a579c1cbef8eb23fe9c3bf7d/src/app/message-reader.h#L35)

Reply:
> The goal of using pointer to the data was to prevent unnecessary stack usage since all of those variables will be pass directly to the callback handler

At the very least, the need to be swapped into native endianness, right?

But also, with the different underlying encoding for CHIP (TLV-based) this won't be viable in any case....